packages/dart_frog_web_socket/lib/src/web_socket_handler.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:dart_frog/dart_frog.dart';
+import 'package:shelf_web_socket/shelf_web_socket.dart' as shelf_web_socket;
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+/// Creates a Dart Frog [Handler] that upgrades HTTP requests to WebSocket
+/// connections.
+///
+/// ```dart
+/// import 'package:dart_frog_web_socket/dart_frog_web_socket.dart';
+///
+/// final onRequest = webSocketHandler(
+///   (channel, protocol) {
+///     // A new connection was established.
+///     print('connected');
+///     // Subscribe to the stream of messages from the client.
+///     channel.stream.listen(
+///       (message) {
+///         // Handle incoming messages.
+///         print('received: $message');
+///         // Send outgoing messages to the connected client.
+///         channel.sink.add('pong');
+///       },
+///       // The connection was terminated.
+///       onDone: () => print('disconnected'),
+///     );
+///   },
+/// );
+/// ```
+///
+/// Only valid WebSocket upgrade requests are upgraded. If a request doesn't
+/// look like a WebSocket upgrade request, a 404 Not Found is returned; if a
+/// request looks like an upgrade request but is invalid, a 400 Bad Request is
+/// returned; and if a request is a valid upgrade request but has an origin that
+/// doesn't match [allowedOrigins] (see below), a 403 Forbidden is returned.
+///
+/// The [onConnection] must take a [WebSocketChannel] as the first argument
+/// and a string for the [WebSocket subprotocol][] as the second
```

Comment:
```suggestion
/// and a string for the [WebSocket subprotocol] as the second
```

Or is it intentional? If it is, curious on why haha

Reply:
It's intentional, the link is further down